### PR TITLE
BACK-1506: Hashflow fix mm bug, add logs

### PR DESCRIPTION
**Pull request by JackieJoo**

### CODE REVIEW

File: src/dex/hashflow/hashflow.ts [hunk 1]
```diff
@@ -714,7 +714,7 @@ export class Hashflow extends SimpleExchange implements IDex<HashflowData> {
     } catch (e) {
       if (
         e instanceof Error &&
-        e.message.endsWith('User is restricted from using Hashflow')
+        e.message.includes('User is restricted from using Hashflow')
```
Comment (mwamedacen) [inline src/dex/hashflow/hashflow.ts:717]: I suggest to lower case message and check just for "user is restricted"

File: src/dex/hashflow/hashflow.ts [hunk 2]
```diff
@@ -726,6 +726,9 @@ export class Hashflow extends SimpleExchange implements IDex<HashflowData> {
             `${this.dexKey}-${this.network}: Market Maker ${mm} failed to build transaction on side ${side} with too strict slippage. Skipping restriction`,
           );
         } else {
+          this.logger.error(
```
Comment (mwamedacen) [inline src/dex/hashflow/hashflow.ts:729]: let's keep it warn (same as most of the errors and [this log](https://github.com/paraswap/paraswap-dex-lib/blob/0b688cd16727616b16c26130adc5d19a15f07792/src/dex/hashflow/hashflow.ts#L741)).

File: src/dex/hashflow/hashflow.ts (hunk 1 above)
Comment (JackieJoo) [reply src/dex/hashflow/hashflow.ts]: done

File: src/dex/hashflow/hashflow.ts (hunk 2 above)
Comment (JackieJoo) [reply src/dex/hashflow/hashflow.ts]: done

**Review by mwamedacen — APPROVED** [review]

**Pull request merged**